support/cas-server-support-x509-webflow/src/main/java/org/apereo/cas/web/extractcert/ExtractX509Certificate.java

```diff
@@ -0,0 +1,25 @@
+package org.apereo.cas.web.extractcert;
+
+import java.security.cert.X509Certificate;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * 
+ * Interface to get an X509 certificate from {@link HttpServletRequest}. e.g.
+ * from pem encoded cert on HTTP header.
+ * 
+ * @author Hal Deadman
+ * @since 5.3.0
+ * 
+ */
+public interface ExtractX509Certificate {

```

Comment:
`X509CertificateExtractor` is a better name.

Also the interface could be tagged as `FunctionalInterface`?